ssl/ssl.go

```diff
@@ -137,13 +149,43 @@ func GenerateCerts(certInfo map[string]CertInfo, namespace, clusterDomain string
 		}(id, info)
 	}
 	wg.Wait()
+
+	var err error
 	if len(errs) > 0 {
-		return errs[0]
+		err = errs[0]
```

Comment:
Yes, this exists only to make sure we delete the outstanding CSRs.

I don't think using `defer` will make this clearer, because it interferes with proper error handling, as you already noted.

However, this now reminds me that I wanted to attempt do delete the CSR before creating it, just in case it somehow escaped proper cleanup in an earlier run.